fdbserver/storageserver.actor.cpp

```diff
@@ -2941,17 +2951,28 @@ ACTOR Future<Void> changeFeedStreamQ(StorageServer* data, ChangeFeedStreamReques
 	state bool atLatest = false;
 	state bool removeUID = false;
 	state Optional<Version> blockedVersion;
-	if (req.replyBufferSize <= 0) {
-		req.reply.setByteLimit(SERVER_KNOBS->CHANGEFEEDSTREAM_LIMIT_BYTES);
-	} else {
-		req.reply.setByteLimit(std::min((int64_t)req.replyBufferSize, SERVER_KNOBS->CHANGEFEEDSTREAM_LIMIT_BYTES));
-	}
 
-	++data->counters.feedStreamQueries;
+	try {
+		++data->counters.feedStreamQueries;
 
-	wait(delay(0, TaskPriority::DefaultEndpoint));
+		// FIXME: do something more sophisticated here besides hard limit
+		if (data->activeFeedQueries >= SERVER_KNOBS->STORAGE_FEED_QUERY_HARD_LIMIT ||
+		    (g_network->isSimulated() && BUGGIFY_WITH_PROB(0.005))) {
+			req.reply.sendError(storage_too_many_feed_streams());
+			++data->counters.rejectedFeedStreamQueries;
+			return Void();
+		}
+
+		data->activeFeedQueries++;
+
+		if (req.replyBufferSize <= 0) {
```

Comment:
Minor nit: this is more style preference, is there a way we can just set replyBufferSize appropriately higher up rather than doing that here? Or is this just how it has to be done. I don't particularly like <= because it acts as a catch all.

Reply:
So unfortunately there isn't a good way to specify defaults across this without a special value for a default.
I agree it's not the best here but was the easiest way to do it at the time